### PR TITLE
Implement tooling best practices

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
https://warthogs.atlassian.net/browse/DPE-3060

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -41,11 +40,12 @@ jobs:
...
         uses: canonical/charming-actions/check-libraries@2.4.0
         with:
-          credentials: "${{ secrets.CHARMHUB_TOKEN }}" # FIXME: current token will expire in 2023-07-04
-          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          credentials: ${{ secrets.CHARMHUB_TOKEN }} # FIXME: current token will expire in 2023-07-04
+          github-token: ${{ secrets.GITHUB_TOKEN }}
 
   build:
     name: Build charms
+    # todo: update version
```
Comment (MiaAltieri) [inline .github/workflows/ci.yaml:48]: [already done]](https://github.com/canonical/mongos-operator/blob/17ddef37a5ca15030b7398b20446b64fd71a5183/.github/workflows/ci.yaml#L49) 🎉

File: charmcraft.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@ bases:
 
 parts:
   charm:
+    # todo: remove binary packages
```
Comment (MiaAltieri) [inline charmcraft.yaml:15]: noob question, why do these need to be removed? last I checked charm wouldn't pack without these

File: tox.ini [hunk 3]
```diff
@@ -35,47 +33,41 @@ commands_post =
...
 description = Apply coding style standards to code
+commands_pre =
+    poetry install --only format
 commands =
-    poetry install
-    poetry export -f requirements.txt -o requirements.txt --without-hashes
-
-    poetry install --only fmt
+    poetry lock --no-update
+    # TODO: remove black?
```
Comment (MiaAltieri) [inline tox.ini:40]: noob question, why remove black

**Review by MiaAltieri — COMMENTED** [review]
amazing work here Carl, your help here is greatly appreciated. Some comments / questions added. :) 

Its great learning from you and your work here

**Review by MiaAltieri — APPROVED** [review]
Thank you so much Carl, this is great

**Review by Mehdi-Bendriss — APPROVED** [review]
Great work, thank you Carl!

**Pull request merged**